screen/contributors/src/main/java/com/ivy/contributors/IvyWalletRepositoryDataSource.kt

```diff
@@ -33,27 +39,59 @@ class IvyWalletRepositoryDataSource @Inject constructor(
         @SerialName("stargazers_count")
         val stars: Int,
         @SerialName("html_url")
-        val url: String
+        val url: String,
     )
 
     companion object {
         private const val CONTRIBUTORS_PER_PAGE = 100
+        private const val DISPLAY_ANONYMOUS_CONTRIBUTORS = true
+        private const val INITIAL_PAGE = 1
     }
 
-    suspend fun fetchContributors(): List<ContributorDto>? {
-        return try {
-            withContext(Dispatchers.IO) {
-                httpClient
-                    .get("https://api.github.com/repos/Ivy-Apps/ivy-wallet/contributors") {
-                        parameter("per_page", CONTRIBUTORS_PER_PAGE)
-                    }
-                    .body<List<ContributorDto>>()
+    suspend fun fetchContributors(): Either<String, List<ContributorDto>> = either {
+        withContext(dispatchersProvider.io) {
+            pagingSource().bind()
+        }
+    }
+
+    private suspend fun pagingSource(): Either<String, List<ContributorDto>> = either {
+        val contributorsSource = mutableListOf<ContributorDto>()
+        var currentPage: Int? = INITIAL_PAGE
+        while (currentPage != null) {
+            val contributorsResult = getContributorsFromRequest(currentPage)
+            contributorsResult.onLeft { errorMessage ->
+                currentPage = null
+                raise(errorMessage)
             }
-        } catch (e: Exception) {
-            null
+            contributorsResult.onRight { results ->
+                currentPage = getNextPage(results, currentPage)
+                currentPage?.let { contributorsSource.addAll(results) }
+            }
+            if (currentPage == null) { break }
         }
+        contributorsSource.toList()
     }
 
+    private suspend fun getContributorsFromRequest(currentPage: Int): Either<String, List<ContributorDto>> =
+        either {
```

Comment:
Remove `either` and return Either.Left() in the catch errro block.  `either {}` is redundant